### PR TITLE
[SPARK-42695][SQL] Skew join handling in stream side of broadcast hash join

### PR DESCRIPTION
**Pull request by xingchaozh**
### What changes were proposed in this pull request?
We could handle the steam side skew of BroadcastHashJoin to improve the join performance

Before | After
-- | --
 | 

### Why are the changes needed?
We can extend the OptimizeSkewedJoin to handle this case

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
UT

### CODE REVIEW

**Comment by wangyum** [comment]
cc @cloud-fan

**Comment by zuston** [comment]
It’s a good improvement, especially for RSS

**Comment by pan3793** [comment]
cc @ulysses-you

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala [hunk 1]
```diff
@@ -215,6 +216,32 @@ case class OptimizeSkewedJoin(ensureRequirements: EnsureRequirements)
...
+      val skewThreshold = getSkewThreshold(medSize)
+      stats.bytesByPartitionId.exists(_ > skewThreshold)
+    }
+
+    if (!shuffleStage.mapStats.forall(isSkewed(_))) {
+      return None
+    }
+
+    Some(SkewJoinChildWrapper(AQEShuffleReadExec(shuffleStage,
+      OptimizeShuffleWithLocalRead.getPartitionSpecs(shuffleStage, None))))
```
Comment (ulysses-you) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala:244]: How can it handle skewed partition ? The code in `OptimizeShuffleWithLocalRead.getPartitionSpecs` does not take reduce partition size into account.

**Comment by mridulm** [comment]
+CC @shardulm94

**Pull request closed without merging**